http/constants.go

```diff
@@ -1,34 +1,106 @@
 package trustlesshttp
 
-import "fmt"
+import (
+	"strconv"
+	"strings"
+)
+
+type ContentTypeOrder string
 
 const (
 	MimeTypeCar                = "application/vnd.ipld.car"            // The only accepted MIME type
 	MimeTypeCarVersion         = "1"                                   // We only accept version 1 of the MIME type
 	FormatParameterCar         = "car"                                 // The only valid format parameter value
 	FilenameExtCar             = ".car"                                // The only valid filename extension
-	DefaultIncludeDupes        = true                                  // The default value for an unspecified "dups" parameter. See https://github.com/ipfs/specs/pull/412.
 	ResponseCacheControlHeader = "public, max-age=29030400, immutable" // Magic cache control values
+	DefaultIncludeDupes        = true                                  // The default value for an unspecified "dups" parameter.
+	DefaultOrder               = ContentTypeOrderDfs                   // The default value for an unspecified "order" parameter.
+
+	ContentTypeOrderDfs ContentTypeOrder = "dfs"
+	ContentTypeOrderUnk ContentTypeOrder = "unk"
```

Comment:
is this a useful value? we don't seem to do anything with an unknown ordering, and if the ordering is anything other than an understood 'dfs' we presumably would want to stream it through unchanged rather than try to change it to "unk"? since we'd be comparing against known values, like 'dfs', what do we get by defining and parsing an 'unk' value here?

Reply:
It's defined by the spec https://specs.ipfs.tech/http-gateways/trustless-gateway/#car-order-content-type-parameter

I'm not particularly a fan of this existing but but the idea of it being in here is that a consumer of this library can decide what to do with it. Currently Lassie will ignore it and assume `dfs` and error on you for feeding bad data. Frisbii will always give you `dfs` regardless of what you ask for (on the assumption that `dfs` is a subset of `unk` so this is acceptable).

This isn't new btw, it was in Lassie like this; no behaviour should be changing with this, it just gets surfaced as a parameter you can inspect now if you care.

Reply:
it does seem like premature complexity / over engineering

Reply:
https://github.com/ipfs/specs/pull/412

It was nearly `dfs`, `rnd`, `weak`!

Our approach atm is to 🙈; although it's at least exposed here if you actually wanted to do something with it. I'm going to assume that nobody will that wants to interoperate with our current tools (Lassie primarily).